### PR TITLE
WIP: Refactor to add magic link support

### PR DESCRIPTION
**Pull request by crashGoBoom**
Issue #HAV-151

Signed-off-by: Christopher Mundus <chris@kindlyops.com>

# Description

Now adds magic link support to the login options.

## Discussion list or pending tasks

* [x] Add base magic link support
* [ ] Adjust registration funnel to use magic links

## How to test
Go to the login page and try to use the magic link option with a current user.

## Checklist
* [x] My pull request is based on the latest master branch
      `git pull --rebase origin master`.
* [x] My code is formatted with elm-format.
* [x] All new and existing tests passed.
* [ ] For visual changes, I've attached before and after screenshots

### CODE REVIEW

File: keycloak/themes/haven/login/login.ftl [hunk 1]
```diff
@@ -54,6 +54,9 @@
             <#if realm.password && social.providers??>
                 <div id="kc-social-providers" class="mb-auto text-center">
                     <p class="mb-0 primary-font">Log In With:</p>
+                        <list>
+                            <p class="mb-1"><a href="/auth/realms/havendev/clients/magic-link/redirect" id="zocial-magic-link" class="zocial magic-link"> <span class="text">Get MagicLink</span></a></p>
```
Comment (statik) [inline keycloak/themes/haven/login/login.ftl]: What do you think about changing this text to "Email me a sign-in link"?

**Comment by statik** [comment]
The apitest is failing in CircleCI but is not showing the keycloak logs - the error is that keycloak is not listening, so I suspect keycloak crashing on startup during plugin load.

Running locally I see this message in the keycloak server logs:
```
keycloak_1            | 18:41:15,033 INFO  [org.jboss.as.server] (Thread-2) WFLYSRV0220: Server shutdown has been requested via an OS signal
keycloak_1            | 18:41:15,043 ERROR [org.jboss.msc.service.fail] (ServerService Thread Pool -- 56) MSC000001: Failed to start service jboss.undertow.deployment.default-server.default-host./auth: org.jboss.msc.service.StartException in service jboss.undertow.deployment.default-server.default-host./auth: java.lang.RuntimeException: RESTEASY003325: Failed to construct public org.keycloak.services.resources.KeycloakApplication(javax.servlet.ServletContext,org.jboss.resteasy.core.Dispatcher)
```
also further down
```
Caused by: java.lang.RuntimeException: Unable to resolve auth flow binding override for: browser
```

File: keycloak/themes/haven/login/login.ftl (hunk 1 above)
Comment (crashGoBoom) [reply keycloak/themes/haven/login/login.ftl]: Good idea! Just changed it

**Comment by crashGoBoom** [comment]
Superseded by #305 Closing out.

**Pull request closed without merging**